src/aof.c

```diff
@@ -2666,15 +2675,6 @@ void backgroundRewriteDoneHandler(int exitcode, int bysignal) {
         /* We can safely let `server.aof_manifest` point to 'temp_am' and free the previous one. */
         aofManifestFreeAndUpdate(temp_am);
 
-        if (server.aof_fd != -1) {
-            /* AOF enabled. */
-            server.aof_selected_db = -1; /* Make sure SELECT is re-issued */
-            server.aof_current_size = getAppendOnlyFileSize(new_base_filename, NULL) + server.aof_last_incr_size;
-            server.aof_rewrite_base_size = server.aof_current_size;
```

Comment:
i think you do need to keep the lines abut the aof_current_size and aof_rewrite_base_size

Reply:
and maybe instead of
```
        if (server.aof_fd != -1) {
            /* AOF enabled. */
```
check `server.aof_state != AOF_OFF`

Reply:
odd this change will casue `WAITAOF replica isn't configured to do AOF` to hang in my local env. i will take a look later

Reply:
ok i see. we update aof_current_size here, it may become smaller since we just done a rewrite
and the next flushAppendOnlyFile write aof, it wil increase the aof_current_size
and then `aof_current_size == aof_fsync_offset`, so we miss the fsync